compiler/parser/RdrHsSyn.hs

```diff
@@ -2574,17 +2584,17 @@ applyConcat (L loc ps) =
 -- | Utility for constructing patterns of the form 'arg@T'.
 asPatPrefixCon :: String -> Located RdrName -> Pat GhcPs
 asPatPrefixCon varName conName =
-  AsPat noExt
+  XPat (noLoc $ AsPat noExt
     (noLoc $ mkRdrUnqual (mkVarOcc varName))
-    (noLoc $ ConPatIn conName $ PrefixCon [])
+    (noLoc $ XPat (noLoc $ ConPatIn conName $ PrefixCon [])))
 
 -- | Utility for constructing patterns of the form 'arg@T{..}'.
 asPatRecWild :: String -> Located RdrName -> Pat GhcPs
 asPatRecWild varName conName =
-  AsPat noExt
+  XPat (noLoc $ AsPat noExt
```

Comment:
Why has this changed?

Reply:
I read it as GHC API changes. It would appear to me that trees that grow extensions are being transitioned into in `data Pat p`. Happened to notice it when I was eye-balling Haskell vs. desugared templates. As noted earlier, have verified that the upgrading does not cause regressions in DAML.